### PR TITLE
Show field caption in query builder

### LINKED ISSUE #2901: Some fields are not showing the proper field caption in query builder results
## Report

> In the screenshot of 7 below the first, 6th and 9th columns have incorrect captions in grod view (not in form view), but the sp6 grid has correct captions.
> 
> I don’t understand why there seems to be a prefix ‘Collection Object’ in front of those captions.
> 
> 
> image
> image

---

## Bug

image

`name` field has the caption "Other ID"

image

Yet, when running a query, the column heading is "Collection Object"

image

In the above screenshots, it is running `v7.7.5`

On our test instance running `edge`, the same issue occurs.

image

[Link to the above query on the test instance](https://sanbibiobank-edge.test.specifysystems.org/specify/query/183/)

---

### Potential Bugs

`fieldNumber` is named "Pathology" in the schema yet the column heading is "Collection Object Pathology":

image

On `edge`, this is not reproducible when selecting `fieldNumber`:

image

---

`catalogedDateVerbatim` is displaying as "Collection Object Markings" instead of "Markings":

image

Cannot recreate in `edge`:

image

---

[Test Deployment](https://sanbibiobank-edge.test.specifysystems.org/specify/query/183/)

**Reported By:** Willem at SAIAB

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixed #2901 

For failing tests, copy only screen shots for errors + error explanation and what should be the expected result.

### CODE REVIEW

**Review by maxxxxxdlp — APPROVED** [review]

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
image

https://sanbibiobank-issue-2901.test.specifysystems.org/specify/query/183/
Username: `kim`
Collection: `Individuals`

This issue is still present on this PR

**Comment by maxxxxxdlp** [comment]
@grantfitzsimmons for the above screenshot, what's are the expected column headings?

**Comment by maxxxxxdlp** [comment]
Edit: oh I see, the "Other ID" column is shown as "Collection Object"

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
image

What is the criteria for prefacing the caption with "Collection Object"?

image

For reference, none of these have "Collection Object" before them in the field caption and they are not relationships.

https://sanbibiobank-issue-2901.test.specifysystems.org/specify/query/183/

**Comment by maxxxxxdlp** [comment]
@grantfitzsimmons the criteria haven't changed since 7.7 so this pull request should not be rejected for that reason. But, feel free to open a new issue.

The current criteria:
If field label is just one word, and the field is not one of the ones listed below, than it is considered generic:

```js
[
  'latitude1',
  'longitude1',
  'latitude2',
  'longitude2',
  'action',
  'fields',
]
```

By generic I mean field label by itself would be ambiguous and table name needs to be included too. The "field label is just one word" rule is very general and exceptions would likely be necessary, though it's a good rule of thumb for a lot of situations. Do you want me to send you the list of all the fields that have a one word label in the default en localization so that you can check if we should mark any of them as not generic?

We can also force some fields to be considered generic. The current list is:
```js
[
  'timestampCreated',
  'timestampModified',
  'createdByAgent',
  'modifiedByAgent',
  'guid',
  'version',
  'id',
]
```

**Comment by grantfitzsimmons** [comment]
Sure! That sounds good. We can approve this PR then, it is working as intended.

**Review by grantfitzsimmons — APPROVED** [review]

File: specifyweb/frontend/js_src/lib/components/WbPlanView/mappingPreview.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ import { getMappingLineData } from './navigator';
 
 /** Use table name instead of field name for the following fields: */
 const fieldsToHide = new Set<string>([
-  'name',
+  // 'name',
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/WbPlanView/mappingPreview.ts:29]: Don't comment it out. Rather, remove it entirely.
Don't want to end up like sp6 with a ton of commented-out code that no one knows what do to with

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]
Tiny change. After that good for merging

**Review by maxxxxxdlp — APPROVED** [review]

**Pull request merged**